dashboard/src/pages/matter-cluster-view.ts

```diff
@@ -0,0 +1,149 @@
+import "@material/web/iconbutton/icon-button";
+import "@material/web/divider/divider";
+import { LitElement, css, html } from "lit";
+import { customElement, property } from "lit/decorators.js";
+import "@material/web/list/list";
+import "@material/web/list/list-item";
+import "@material/web/divider/divider";
+import { MatterClient } from "../client/client";
+import "../components/ha-svg-icon";
+import { MatterNode } from "../client/models/node";
+import { clusters } from "../client/models/descriptions";
+import { showAlertDialog } from "../components/dialog-box/show-dialog-box";
+
+declare global {
+  interface HTMLElementTagNameMap {
+    "matter-cluster-view": MatterClusterView;
+  }
+}
+
+function clusterAttributes(attributes: { [key: string]: any }, endpoint: number, cluster: number) {
+  // extract unique clusters from the node attributes, as (sorted) array
+  const endpointClusters = Object.keys(attributes)
+    .filter(key => key.startsWith(`${endpoint}/${cluster}`))
+    .map((key) => {
+      const attributeKey = Number(key.split('/')[2]);
+      return { key: attributeKey, value: attributes[key] };
+    }, []);
+  return endpointClusters;
+}
+
+@customElement("matter-cluster-view")
+class MatterClusterView extends LitElement {
+  public client!: MatterClient;
+
+  @property()
+  public node?: MatterNode;
+
+  @property()
+  public endpoint?: number;
+
+  @property()
+  public cluster?: number;
+
+  render() {
+
+    if (!this.node || this.endpoint == undefined || this.cluster == undefined) {
+      return html`
+        <p>Node, endpoint or cluster not found!</p>
+        <button
+          @click=${() => {
+          history.back();
+        }}
```

Comment:
Dont define event handlers in your template, this means they will be recreated on every render

Reply:
fixed